raiden/network/proxies/secret_registry.py

```diff
@@ -142,6 +142,9 @@ def check_registered(
             secrethash: SecretHash,
             block_identifier: BlockSpecification,
     ) -> bool:
+        if not self.client.can_query_state_for_block(block_identifier):
+            block_identifier = 'latest'
```

Comment:
IMO the block identifier should not be silently changed.

Reply:
so what is the alternative? Return false? Return true?

Reply:
raise an exception

Reply:
But why? This can happen. It's valid behavior. If a secret registration is in our pending transactions and we shut down and then restart after 128 blocks we should be able to function normally.

Reply:
> But why?

Because silently changing behavior makes understanding code harder.

> This can happen

It should not. Client code `check_registered(secret_hash, extremely_old_block)` is wrong, it should not do this call.

> It's valid behavior

If it was valid behavior to call the function with an old block number, we wouldn't need the special case.

>  If a secret registration is in our pending transactions and we shut down and then restart after 128 blocks we should be able to function normally.

I didn't say the node should crash.

Reply:
Raised an exception instead, and handle it at the place where this can happen